pkg/cloud/services/gc/ec2.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+func (s *Service) deleteSecurityGroups(ctx context.Context, resources []*AWSResource) error {
+	for _, resource := range resources {
+		if !s.isSecurityGroupToDelete(resource) {
+			s.scope.V(4).Info("Resource not a security group for deletion", "arn", resource.ARN.String())
+			continue
+		}
+
+		groupID := strings.ReplaceAll(resource.ARN.Resource, "security-group/", "")
+		if err := s.deleteSecurityGroup(ctx, groupID); err != nil {
+			return fmt.Errorf("deleting security group %s: %w", groupID, err)
+		}
+	}
+	s.scope.V(2).Info("Finished processing resources for security group deletion")
+
+	return nil
+}
+
+func (s *Service) isSecurityGroupToDelete(resource *AWSResource) bool {
+	//TODO: should we check for the security group name start with k8s-elb-
+
+	if resource.ARN.Service != ec2.ServiceName {
+		s.scope.V(5).Info("Resource is not for ec2 service", "arn", resource.ARN.String(), "check", "securitygroup")
+		return false
+	}
+	if !strings.HasPrefix(resource.ARN.Resource, "security-group/") {
+		s.scope.V(5).Info("Resource is not a security group", "arn", resource.ARN.String(), "check", "securitygroup")
+		return false
+	}
+	eksClusterName := resource.Tags[eksClusterNameTag] //nolint:ifshort
```

Comment:
Why the nolint? :D if you don't use this value outside the if why not inline it?

Reply:
I'm surprised. In my experience, if a nolint directive is not needed, the linter complains!